BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m

```diff
@@ -14,6 +14,7 @@
 #import "BOXAppUserSession.h"
 #import "NSString+BOXContentSDKAdditions.h"
 #import "UIDevice+BOXContentSDKAdditions.h"
+#import "BOXContentClient.h"
```

Comment:
I'm not sure we'd do things differently.
I guess one option would be to try and make the URL-switching happen around the request object perhaps at the 'prepareRequest'. That way there is still an awareness of the Request in the Client, but not the awareness of the Client in the Request. Not sure if that is feasible or if so, that's it's definitely better. I'll leave it to you.